src/support/modify.c

```diff
@@ -447,8 +447,8 @@ __wt_modify_apply_api(WT_CURSOR *cursor, WT_MODIFY *entries, int nentries)
  *     Takes an in-memory modify and populates an update value with the reconstructed full value.
  */
 int
-__wt_modify_reconstruct_from_upd_list(
-  WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_UPDATE *modify, WT_UPDATE_VALUE *upd_value)
+__wt_modify_reconstruct_from_upd_list(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt,
+  WT_UPDATE *modify, WT_UPDATE_VALUE *upd_value, bool reconciliation)
```

Comment:
A little documentation on the new parameter would be nice - without the context of the change, it's a little unclear - e.g. does this mean the call will do reconciliation?

Reply:
Fixed as per discussion

Reply:
I'd like to offer a suggestion, consider:
```
__wt_modify_reconstruct_from_upd_list(session, cbt, modify, upd_value, false);
```
The meaning of the last parameter is unclear when skimming/reviewing the code, in a larger context. For instance I can see `false` being read as "don't actually modify the update list" -- rather see if it would be successful, that is: a dry run.

Instead if you use an enumeration, while it might feel onerous, it can add clarity in a larger context:
```
/* Illustration only. */
typedef enum { WT_OPCTX_TRANSACTION, WT_OPCTX_RECONCILATION }  WT_OP_CONTEXT;

__wt_modify_reconstruct_from_upd_list(session, cbt, modify, upd_value, WT_OPCTX_TRANSACTION);
``` 
Above I think the intention is clear and the reader does not need to remember the signature.

Reply:
I like this idea - thanks for the suggestion @marcbutler . @jiechenbo what do you reckon?

Reply:
Thank you, i really like that idea! I have incorporated that change here. [bcb782c](https://github.com/wiredtiger/wiredtiger/pull/10311/commits/bcb782c2a9519c6debb93acf0e6fa5d29e294ba3)